diff --git a/node_modules/expo-print/android/src/main/java/expo/modules/print/PrintModule.kt b/node_modules/expo-print/android/src/main/java/expo/modules/print/PrintModule.kt
index 719d296..05fc743 100644
--- a/node_modules/expo-print/android/src/main/java/expo/modules/print/PrintModule.kt
+++ b/node_modules/expo-print/android/src/main/java/expo/modules/print/PrintModule.kt
@@ -19,9 +19,13 @@ import java.lang.ref.WeakReference
 import kotlin.coroutines.Continuation
 import kotlin.coroutines.resume
 import kotlin.coroutines.resumeWithException
+import kotlin.math.roundToInt
 
 private const val ORIENTATION_PORTRAIT = "portrait"
 private const val ORIENTATION_LANDSCAPE = "landscape"
+private const val PRINT_PIXELS_PER_INCH = 72
+private const val PRINT_MILS_PER_INCH = 1000
+private const val PRINT_PIXELS_PER_MIL = PRINT_PIXELS_PER_INCH.toFloat() / PRINT_MILS_PER_INCH
 
 class PrintModule : Module() {
   private val jobName = "Printing"
@@ -155,14 +159,29 @@ class PrintModule : Module() {
   private fun getAttributesFromOptions(options: PrintOptions): PrintAttributes.Builder {
     val orientation = options.orientation
     val builder = PrintAttributes.Builder()
-
-    // @tsapeta: Unfortunately these attributes might be ignored on some devices or Android versions,
-    // in other words it might not change the default orientation in the print dialog,
-    // however the user can change it there.
-    if (ORIENTATION_LANDSCAPE == orientation) {
-      builder.setMediaSize(PrintAttributes.MediaSize.UNKNOWN_LANDSCAPE)
+    val width = options.width
+    val height = options.height
+
+    if (width != null && height != null) {
+      var mediaSize = PrintAttributes.MediaSize(
+        "expo_custom_${width}x${height}",
+        "Personalizado",
+        (width / PRINT_PIXELS_PER_MIL).roundToInt(),
+        (height / PRINT_PIXELS_PER_MIL).roundToInt()
+      )
+      if (ORIENTATION_LANDSCAPE == orientation) {
+        mediaSize = mediaSize.asLandscape()
+      }
+      builder.setMediaSize(mediaSize)
     } else {
-      builder.setMediaSize(PrintAttributes.MediaSize.UNKNOWN_PORTRAIT)
+      // @tsapeta: Unfortunately these attributes might be ignored on some devices or Android versions,
+      // in other words it might not change the default orientation in the print dialog,
+      // however the user can change it there.
+      if (ORIENTATION_LANDSCAPE == orientation) {
+        builder.setMediaSize(PrintAttributes.MediaSize.UNKNOWN_LANDSCAPE)
+      } else {
+        builder.setMediaSize(PrintAttributes.MediaSize.UNKNOWN_PORTRAIT)
+      }
     }
 
     // @tsapeta: It should just copy the document without adding extra margins,
